graph/history.go

```diff
@@ -8,6 +9,78 @@ import (
 	"github.com/docker/docker/utils"
 )
 
+// History returns the list of all images used to create this image.
+func (graph *Graph) History(img *image.Image) ([]*image.Image, error) {
```

Comment:
Not sure if this is relevant: what is the advantage of the pointer types for return and `WalkHistory` callbacks? It feels like we are exposing unprotected memory without proper locking. Yes, it is less memory efficient, but that may be a premature optimization. The output of history is small and the callback would nearly be all on stack if you're careful.